deps/wazuh_testing/wazuh_testing/authd.py

```diff
@@ -0,0 +1,197 @@
+        command = command + f" IP:'{ip}'"
+
+    if 'key_hash' in input:
+        key_hash = input['key_hash']
+        command = command + f" K:'{key_hash}'"
+
+    return command
+
+
+# Functions
+def validate_authd_logs(expected_logs, log_monitor=None):
+    if not log_monitor:
+        log_monitor = FileMonitor(LOG_FILE_PATH)
+
+    for log in expected_logs:
+        log_monitor.start(timeout=AUTHD_KEY_REQUEST_TIMEOUT,
+                          callback=make_callback(log, prefix=AUTHD_DETECTOR_PREFIX,
+                                                 escape=True),
+                          error_message=f"Expected log does not occured: '{log}'")
+
+
+def validate_argument(received, expected, argument_name):
+    if received != expected:
+        return 'error', f"Invalid '{argument_name}': '{received}' received, '{expected}' expected."
+    else:
+        return 'success', ''
+
+
+def validate_authd_response(response, expected):
+    """
+    Validates if the different items of an Authd response are as expected. Any item inexistent in expected won't
+    be validated.
+
+    Args:
+        response (str): The Authd response to be validated.
+        expected (dict): Dictionary with the items to validate.
+    """
+    response = response.split(sep=" ", maxsplit=1)
+    status = response[0]
+    result = 'success'
+    err_msg = ''
+    if expected['status'] == 'success':
+        result, err_msg = validate_argument(status, 'OSSEC', 'status')
+        if result != 'success':
+            return result, err_msg
+
+        agent_key = response[1].split('\'')[1::2][0].split()
+        id = agent_key[0]
+        name = agent_key[1]
+        ip = agent_key[2]
+        key = agent_key[3]
+
+        if 'id' in expected:
+            result, err_msg = validate_argument(id, expected['id'], 'id')
+            if result != 'success':
+                return result, err_msg
+
+        if 'name' in expected:
+            result, err_msg = validate_argument(name, expected['name'], 'name')
+            if result != 'success':
+                return result, err_msg
+
+        if 'ip' in expected:
+            result, err_msg = validate_argument(ip, expected['ip'], 'ip')
+            if result != 'success':
+                return result, err_msg
+
+        if 'key' in expected:
+            result, err_msg = validate_argument(key, expected['key'], 'key')
+            if result != 'success':
+                return result, err_msg
+
+    elif expected['status'] == 'error':
+        result, err_msg = validate_argument(status, 'ERROR:', 'status')
+        if result != 'success':
+            return result, err_msg
+
+        message = response[1]
+        if 'message' in expected:
+            if re.match(expected['message'], message) is None:
+                return 'error', f"Invalid 'message': '{message}' received, '{expected['message']}' expected"
+    else:
+        raise Exception('Invalid expected status')
+
+    return result, err_msg
```

Comment:
This function is a little hard to read and it has code repetition. In future revisions of this module we could use something like the following:

```suggestion
def get_authd_response_values(response):
    authd_response_dict = {}
    try: 
        response = response.split(sep=' ', maxsplit=1)
        status, message = response[0:1]
        
        authd_response_dict['status'] = status
        authd_response_dict['message'] = message
        
        if status == 'success':
            agent_key = message.split('\'')[1::2][0].split()
            id, name, ip, key = agent_key[0:4]
            authd_response_dict['id'] = id
            authd_response_dict['name'] = name
            authd_response_dict['ip'] = ip
            authd_response_dict['key'] = key
    except:
        raise IndexError(f"Authd response {response} does not contains expected values: {str(values)}")
    return authd_response_dict
    
        
def validate_authd_response(response, expected):
    """
    Validates if the different items of an Authd response are as expected. Any item inexistent in expected won't
    be validated.

    Args:
        response (str): The Authd response to be validated.
        expected (dict): Dictionary with the items to validate.
    """
    authd_response_dict = get_authd_response_values(response)

    result = 'success'
    err_msg = ''
    
    for expected_key in expected.keys():
        if expected_key == 'message':
            if re.match(expected[expected_key], authd_response_dict[message]) is None:
                return 'error', f"Invalid 'message': '{message}' received, '{expected['message']}' expected"
        elif expected_key in ['status', 'id', 'name', 'ip', 'key']:
            if expected_key == 'status':
                expected_value = 'OSSEC' if expected['status'] == 'success' else 'ERROR'
            else:
                expected_value = expected[expected_key]
            
            result, err_msg = validate_argument(authd_response_dict[expected_value], expected_value, expected_key)
        else:
            raise Exception('Invalid expected status')

    return result, err_msg
```